### PR TITLE
Document installation procedure for Frontier

### PR DESCRIPTION
**Pull request by balos1**
RE https://github.com/LLNL/sundials-private/issues/58. Add a section to the install guide on how to build sundials on Frontier.

### CODE REVIEW

**Review by drreynolds — DISMISSED** [review]
This looks great.  I have only a couple of very minor changes.

File: doc/shared/sundials/Install.rst [hunk 1]
```diff
@@ -2112,3 +2112,68 @@ Table of SUNDIALS libraries and header files
...
+Leadership Computing Facility. If you are new to this system, then we recommend that you review the
+`Frontier user guide <https://docs.olcf.ornl.gov/systems/frontier_user_guide.html>`_.
+
+**A Standard Installation**
+
+Clone SUNDIALS:
+
+```bash
+git clone https://github.com/LLNL/sundials.git && cd sundials
+```
```
Comment (gardner48) [inline doc/shared/sundials/Install.rst:2141]:
```suggestion
.. code-block:: bash

   git clone https://github.com/LLNL/sundials.git && cd sundials
```
```

File: doc/shared/sundials/Install.rst [hunk 2]
```diff
@@ -2112,3 +2112,68 @@ Table of SUNDIALS libraries and header files
...
+export MPICH_GPU_SUPPORT_ENABLED=1
+
+# compiler environment hints
+export CC=$(which hipcc)
+export CXX=$(which hipcc)
+export FC=$(which ftn)
+export CFLAGS="-I${ROCM_PATH}/include"
+export CXXFLAGS="-I${ROCM_PATH}/include -Wno-pass-failed"
+export LDFLAGS="-L${ROCM_PATH}/lib -lamdhip64 ${PE_MPICH_GTL_DIR_amd_gfx90a} -lmpi_gtl_hsa"
+```
```
Comment (gardner48) [inline doc/shared/sundials/Install.rst:2165]:
```suggestion
.. code-block:: bash

   # required dependencies
   module load PrgEnv-cray-amd/8.5.0
   module load craype-accel-amd-gfx90a
   module load rocm/5.3.0
   module load cmake/3.23.2

   # GPU-aware MPI
   export MPICH_GPU_SUPPORT_ENABLED=1

   # compiler environment hints
   export CC=$(which hipcc)
   export CXX=$(which hipcc)
   export FC=$(which ftn)
   export CFLAGS="-I${ROCM_PATH}/include"
   export CXXFLAGS="-I${ROCM_PATH}/include -Wno-pass-failed"
   export LDFLAGS="-L${ROCM_PATH}/lib -lamdhip64 ${PE_MPICH_GTL_DIR_amd_gfx90a} -lmpi_gtl_hsa"
```

File: doc/shared/sundials/Install.rst [hunk 3]
```diff
@@ -2112,3 +2112,68 @@ Table of SUNDIALS libraries and header files
...
+```bash
+cmake -S . -B builddir -DCMAKE_INSTALL_PREFIX=<install path> -DAMDGPU_TARGETS=gfx90a \
+   -DENABLE_HIP=ON -DENABLE_MPI=ON -DBUILD_FORTRAN_MODULE_INTERFACE=ON
+cd builddir
+make -j8 install
+# Need an allocation to run the tests:
+salloc -A <account> -t 10 -N 1 -p batch
+make test
+make test_install_all
+```
```
Comment (gardner48) [inline doc/shared/sundials/Install.rst:2179]:
```suggestion
.. code-block:: bash

   cmake -S . -B builddir -DCMAKE_INSTALL_PREFIX=<install path> -DAMDGPU_TARGETS=gfx90a \
   -DENABLE_HIP=ON -DENABLE_MPI=ON -DBUILD_FORTRAN_MODULE_INTERFACE=ON
   cd builddir
   make -j8 install
   # Need an allocation to run the tests:
   salloc -A <account> -t 10 -N 1 -p batch
   make test
   make test_install_all
```

File: doc/shared/sundials/Install.rst [hunk 4]
```diff
@@ -2112,3 +2112,68 @@ Table of SUNDIALS libraries and header files
...
+export CXXFLAGS="-I${ROCM_PATH}/include -Wno-pass-failed"
+export LDFLAGS="-L${ROCM_PATH}/lib -lamdhip64 ${PE_MPICH_GTL_DIR_amd_gfx90a} -lmpi_gtl_hsa"
+```
+
+Now we can build SUNDIALS. In general, this is the same procedure described in the previous sections.
+The following command builds and install SUNDIALS with MPI, HIP, and the Fortran interface enabled:
+
+```bash
+cmake -S . -B builddir -DCMAKE_INSTALL_PREFIX=<install path> -DAMDGPU_TARGETS=gfx90a \
+   -DENABLE_HIP=ON -DENABLE_MPI=ON -DBUILD_FORTRAN_MODULE_INTERFACE=ON
```
Comment (gardner48) [inline doc/shared/sundials/Install.rst:2172]: Does CMake pickup `srun` as the `MPIEXEC_EXECUTABLE` automatically for running the tests? Previously this was not the case and it needed to be set explicitly.

**Review by gardner48 — CHANGES_REQUESTED** [review]
Minor fixes for formatting code blocks and one question

File: doc/shared/sundials/Install.rst (hunk 4 above)
Comment (balos1) [reply doc/shared/sundials/Install.rst:2172]: Yes, it picks up `srun` as the `MPIEXEC_EXECUTABLE`.

File: doc/shared/sundials/Install.rst (hunk 1 above)
Comment (balos1) [reply doc/shared/sundials/Install.rst:2141]: I forgot this was not markdown clearly :)

**Review by gardner48 — APPROVED** [review]

**Pull request merged**